crates/rpc/rpc-types/src/eth/pubsub.rs

```diff
@@ -76,11 +83,11 @@ pub enum SubscriptionKind {
     Logs,
     /// New Pending Transactions subscription.
     ///
-    /// Returns the hash for all transactions that are added to the pending state and are signed
-    /// with a key that is available in the node. When a transaction that was previously part of
-    /// the canonical chain isn't part of the new canonical chain after a reorganization its again
-    /// emitted.
-    NewPendingTransactions,
+    /// Returns the hash or full tx for all transactions that are added to the pending state and
+    /// are signed with a key that is available in the node. When a transaction that was
+    /// previously part of the canonical chain isn't part of the new canonical chain after a
+    /// reorganization its again emitted.
+    NewPendingTransactions(bool),
```

Comment:
this should be moved to the params object instead

Reply:
Why should this variant be moved to the params object, when we use it to check the kind of subscription?

Reply:
because this enum (NewPendingTransactions) is the first argument of the rpc call, the second argument is `Params` which will contain additional arguments:

https://github.com/paradigmxyz/reth/blob/f3d7988f684d087a895d288a0041e95a6596b8a6/crates/rpc/rpc/src/eth/pubsub.rs#L80-L81

params: [`newPendindingTransactions`, true] will match kind, params